### ISSUE TITLE
Consider exporting ResourceFunc and ResourceInput

### ISSUE THREAD

**Issue by kgilden** [fullcalendar/fullcalendar#5797]
### Checklist

Please mark these items with an [x]

  - [x] I've already searched through [existing tickets](https://fullcalendar.io/issues)
  - [x] Other people will find this feature useful

### Feature Description

In v4 both `ResourceFunc` and `ResourceInput` were exported and users could depend on these types. In v5 it seems that these two types didn't make it to the export list. A good reason for also exporting would be symmetry: `EventSourceInput` and `EventSourceFunc` are exported so as a user I would expect the same for resources. Would you consider exporting them?

If you're ok with this, I would gladly make a PR to [fullcalendar-schedule](https://github.com/fullcalendar/fullcalendar-scheduler) as this is a trivial change.

**Comment by kgilden** [comment]
As a workaround the following works.

```
import { RawOptionsFromRefiners } from '@fullcalendar/core';
import { ResourceSourceInput } from '@fullcalendar/resource-common';

type Unpacked<T> = T extends (infer U)[] ? U : T;

/**
 * The `ResourceFunc` type is not exported since v5 of fullcalendar. This
 * hack here works around that by extracting the `ResourceFunc`.
 */
export type ResourceFunc = Extract<ResourceSourceInput, (...args: any) => any>;

/**
 * The `ResourceInput` type is not exported since v5 of fullcalendar. This
 * hack here works around that by extracting the `ResourceInput`.
 */
export type ResourceInput = Unpacked<Exclude<ResourceSourceInput, string | ResourceFunc | RawOptionsFromRefiners<any>>>;
```

**Comment by arshaw** [comment]
this is implemented in 5.4.0 !

**Comment by kgilden** [comment]
Awesome! Thank you very much!
